app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/vm.rb

```diff
@@ -112,15 +112,82 @@ def raw_destroy
     unsupported_reason_add(:resize, _('The VM is not powered off')) unless current_state == "off"
   end
 
-  def raw_resize(new_flavor)
+  def raw_resize(options)
     with_provider_object do |instance|
-      instance.resize(instance, new_flavor.name)
+      instance.resize(instance, options["name"])
     rescue IBMCloudSdkCore::ApiException => e
       Notification.create(:type => :vm_resize_error, :options => {:subject => instance[:name], :error => e.error})
       raise MiqException::MiqProvisionError, e.to_s
     end
   end
 
+  def params_for_resize
+    {
+      :fields => [
+        {
+          :component  => 'select',
+          :name       => 'flavor',
+          :id         => 'flavor',
+          :label      => _('Flavor'),
+          :isRequired => true,
+          :options    => resize_form_options(408).map do |flavor|
+            {
+              :label => flavor[:name],
+              :value => flavor[:id],
+            }
+          end,
+        },
+      ],
+    }
+  end
+
+  def resize_form_options(recordId)
+    # assert_privileges("instance_resize")
+
+    # @request_id = params[:id]
+    # @record = find_record_with_rbac(VmOrTemplate, recordId)
+    @record = ManageIQ::Providers::IbmCloud::VPC::CloudManager::Vm.find_by(:id => 408)
```

Comment:
This is an instance method that is called from a vm object, so you don't need to lookup the record manually.